tesseract-server/src/handlers/logic_layer/geoadjacents.rs

```diff
@@ -0,0 +1,351 @@
+            None => continue,
+        };
+        for cut in cut_key_values{
+            for op in &operations{
+                let mut level_name = match cube_cache.dimension_caches.get(cut_key){
+                    Some(dimension_cache) => {
+                        match dimension_cache.id_map.get(&cut) {
+                            Some(level_name) => {
+                                if level_name.len() > 1 {
+                                    return Err(format_err!("{} matches multiple levels in this dimension.", cut))
+                                }
+                                match level_name.get(0) {
+                                    Some(ln) => ln.clone(),
+                                    None => return Err(format_err!("{} matches no levels in this dimension.", cut))
+                                }
+                            },
+                            None => continue
+                        }
+                    },
+                    None => {
+                        match level_map.get(cut_key) {
+                            Some(level_name) => {
+                                level_matches.push(level_name.clone());
+                                level_name.clone()
+                            },
+                            None => continue,
+                        }
+                    }
+                };
+
+                if op.to_string() == "children".to_string() {
+                    let child_level = match cube.get_child_level(&level_name)? {
+                        Some(child_level) => child_level,
+                        None => continue  // This level has no child
+                    };
+
+                    let child_level_name = LevelName {
+                        dimension: level_name.dimension.clone(),
+                        hierarchy: level_name.hierarchy.clone(),
+                        level: child_level.name.clone()
+                    };
+
+
+                    // Get children IDs from the cache
+                    let level_cache = match cube_cache.level_caches.get(&level_name.level) {
+                        Some(level_cache) => level_cache,
+                        None => return Err(format_err!("Could not find cached entries for {}.", level_name.level))
+                    };
+
+                    let children_ids = match &level_cache.children_map {
+                        Some(children_map) => {
+                            match children_map.get(&cut) {
+                                Some(children_ids) => children_ids.clone(),
+                                None => continue
+                            }
+                        },
+                        None => continue
+                    };
+
+                    for children_id in children_ids.iter(){
+                        dimensions_map.push([level_name.level.to_string(), cut.to_string(), "children".to_string(), children_id.to_string()].to_vec())
```

Comment:
I don't think this will work well for the level names. We need to return their unique names according to the logic layer, not the name of the level in their level name object. These two will not always be the same. Let me know if you need any clarification.